### PR TITLE
feat: add-user/update-user API should check if username/id/email/phone has duplicated with existing user

### PR DESCRIPTION
**Pull request by AtCliffUnderline**
Fix: https://github.com/casdoor/casdoor/issues/3293

Not sure should it be in controller or better to move to `check.go`?

### CODE REVIEW

**Comment by AtCliffUnderline** [comment]
> @AtCliffUnderline fix CI:
> 
> image

done thx

**Comment by hsluoyz** [comment]
@dacongda plz review

File: controllers/user.go [hunk 1]
```diff
@@ -375,6 +376,20 @@ func (c *ApiController) AddUser() {
 		return
 	}
 
+	if user.Phone != "" {
```
Comment (dacongda) [inline controllers/user.go:379]: imho, same code should be apply in UpdateUser controller maybe you should pack these code as  a function in check.go name `CheckUserEmailAndPhone`

Comment (AtCliffUnderline) [reply controllers/user.go:379]: object/check.go:532

These checks are already implemented in UpdateUser, however first level `if` is not same, so i am confused how to re-use this code in any other way.

Comment (dacongda) [reply controllers/user.go:379]: emmmm, maybe we can create a empty user object and pass it to function `CheckUpdateUser`

Comment (dacongda) [reply controllers/user.go:379]: Example
```go
func (c *ApiController) AddUser() {
	var user object.User
	err := json.Unmarshal(c.Ctx.Input.RequestBody, &user)
	if err != nil {
		c.ResponseError(err.Error())
		return
	}

	count, err := object.GetUserCount("", "", "", "")
	if err != nil {
		c.ResponseError(err.Error())
		return
	}

	if err := checkQuotaForUser(int(count)); err != nil {
		c.ResponseError(err.Error())
		return
	}

	emptyUser := object.User{}
	if msg := object.CheckUpdateUser(&emptyUser, &user, c.GetAcceptLanguage()); msg != "" {
		c.ResponseError(msg)
		return
	}

	c.Data["json"] = wrapActionResponse(object.AddUser(&user))
	c.ServeJSON()
}
```

Comment (AtCliffUnderline) [reply controllers/user.go:379]: Well, if you think this is OK solution

**Review by dacongda — APPROVED** [review]

**Pull request merged**